src/pyOpenMS/pyopenms/_dataframes.py

```
@@ -652,87 +652,34 @@ def _add_meta_values(df: _pd.DataFrame, object: any) -> _pd.DataFrame:
     mvs = []
     object.getKeys(mvs)
+    cnt = df.shape[0]
+
     for k in mvs:
+        k_str = k.decode()
+        if not object.metaValueExists(k):
+            continue
+
         v = object.getMetaValue(k)
-        dtype = 'U100'
+        if isinstance(v, DataValue):
```

Ruff (0.8.2)

Undefined name `DataValue`

(F821)